### DISCUSSION TITLE
BKtool  and auto resitance

### DISCUSSION THREAD

**Discussion by fcna** [cagnulein/qdomyos-zwift#535] [General]
Hello,

First, thanks for the application. Working great with Zwit.

I want to try with bkool. Everything is working (ppm, Bpm) exept resistance. 

Do you have any sugestion ?

By advance thanks

**Comment by cagnulein** [comment]
it should work as Zwift. Could you append here a debug log?

Roberto Viola
Software engineer and open source enthusiast
http://robertoviola.cloud



Il giorno mer 29 dic 2021 alle ore 12:11 fcna ***@***.***> ha
scritto:

**Comment by fcna** [comment]
Thanks, 


log 

[debug-Wed_Dec_29_12_53_32_2021.log](https://github.com/cagnulein/qdomyos-zwift/files/7788822/debug-Wed_Dec_29_12_53_32_2021.log)
[debug-Wed_Dec_29_13_10_54_2021.log](https://github.com/cagnulein/qdomyos-zwift/files/7788823/debug-Wed_Dec_29_13_10_54_2021.log)
[debug-Wed_Dec_29_13_13_41_2021.log](https://github.com/cagnulein/qdomyos-zwift/files/7788824/debug-Wed_Dec_29_13_13_41_2021.log)

[debug-Wed_Dec_29_13_16_30_2021.log](https://github.com/cagnulein/qdomyos-zwift/files/7788821/debug-Wed_Dec_29_13_16_30_2021.log)

and a screen where all seems Ok ...

![PXL_20211229_121724823](https://user-images.githubusercontent.com/43406783/147662264-0a0d40de-26dc-4e81-b17a-21402bcf336b.jpg)

**Reply by cagnulein** [reply]
in this log debug-Wed_Dec_29_12_53_32_2021.log

i see correctly the resistance is changing accordingly to the request of the software

mer. déc. 29 13:02:54 2021 1640779374688 Debug: ../src/domyosbike.cpp void domyosbike::update() "writing resistance 8"
mer. déc. 29 13:02:55 2021 1640779375577 Debug: ../src/domyosbike.cpp void domyosbike::update() "writing resistance 8"
mer. déc. 29 13:03:46 2021 1640779426828 Debug: ../src/domyosbike.cpp void domyosbike::update() "writing resistance 7"
mer. déc. 29 13:03:48 2021 1640779428634 Debug: ../src/domyosbike.cpp void domyosbike::update() "writing resistance 12"
mer. déc. 29 13:03:55 2021 1640779435804 Debug: ../src/domyosbike.cpp void domyosbike::update() "writing resistance 11"
mer. déc. 29 13:03:56 2021 1640779436414 Debug: ../src/domyosbike.cpp void domyosbike::update() "writing resistance 11"

so maybe i don't understand which is the issue?

**Comment by fcna** [comment]
I sent all my log (with diffèrent apps) . Take lthe last log debug-Wed_Dec_29_13_16_30_2021 for BKtool. there is no "writing resistance 8" ...

**Reply by cagnulein** [reply]
in that log bkool app is sending only 

mer. déc. 29 13:17:13 2021 1640780233010 Debug: ../src/virtualbike.cpp void virtualbike::characteristicChanged(const QLowEnergyCharacteristic &, const QByteArray &) "characteristicChanged 10969 00"
mer. déc. 29 13:17:13 2021 1640780233047 Debug: ../src/virtualbike.cpp void virtualbike::characteristicChanged(const QLowEnergyCharacteristic &, const QByteArray &) "characteristicChanged 10969 07"

mean request access and start simulation. On both commands QZ answers with a ACK. But after that bkool doesn't send anything. I guess you should write to the bkool support and put me in CC

Let me know

**Comment by fcna** [comment]
Done. I write to the bkool support but not possible to put you in CC. I will transfert all the informations.

Thanks

**Reply by cagnulein** [reply]
ok! let me know, i'm really interested about this. Maybe you can add me to the conversation in the next followups :)

**Comment by fcna** [comment]
Yes of course

**Comment by fcna** [comment]
Another question. In zwift, I can't connect the BPM infomation. For all other information, it's working but the never found my BPM sensor ...
I m it's working before, and  I think I read i must wait for BPM but after one minutes it's not working ...

**Reply by cagnulein** [reply]
is your android a > 9? if so it's a android limitation

**Comment by fcna** [comment]
Yes Pixel 6, so past version ...

**Reply by cagnulein** [reply]
so it's bigger than version 9?

**Reply by fcna** [reply]
sorry, Android 12; last version
